hyperlink/test/test_url.py

```diff
@@ -231,9 +254,21 @@ def test_roundtrip(self):
         L{URL.to_text} should invert L{URL.from_text}.
         """
         for test in ROUNDTRIP_TESTS:
-            result = URL.from_text(test).to_text()
+            result = URL.from_text(test).to_text(with_password=True)
             self.assertEqual(test, result)
 
+    def test_roundtrip_double_iri(self):
```

Comment:
This looks like a place where it might be interesting to drop in hypothesis (since it was in fact hypothesis that caught this bug, in txacme's test suite)